wp-cache.php

```diff
@@ -3672,6 +3686,30 @@ function wpsc_enable_preload() {
 	wp_schedule_single_event( time() + 10, 'wp_cache_full_preload_hook' );
 }
 
+function wpsc_get_post_types() {
+
+	$preload_type_args = apply_filters( 'wpsc_preload_type_args', array(
+		'public'             => true,
+		'publicly_queryable' => true
+	) );
+
+	$post_types = (array) apply_filters( 'wpsc_preload_post_types', get_post_types( $preload_type_args, 'names', 'or' ));
+
+	return join( "', '", array_map( 'esc_sql', $post_types ) );
+}
+function wpsc_post_count() {
+	global $wpdb;
+	static $count;
+
+	if ( isset( $count ) ) {
+		return $count;
+	}
+
+	$post_type_list = wpsc_get_post_types();
+	$count = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM {$wpdb->posts} WHERE post_type IN ('$post_type_list') AND post_status = 'publish'", $wpdb->posts ) );
```

Comment:
I optimized query (see PHP notice):

`$count = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM $wpdb->posts WHERE post_type IN ($post_type_list) AND post_status = %s", 'publish' ) );`
I moved `'` into function `wpsc_get_post_types`. Also `prepare` requires at least one `%s` but we can't use `$post_type_list`. So, add `%s` or remove `prepare`?